### DISCUSSION TITLE
Pdf Resolution

### DISCUSSION THREAD

**Discussion by ntec1** [ImageMagick/ImageMagick#5079] [Help]
I am using imagemagick and installed ghostscript on windows to use with pdf files. For some reason when using pdf files, the pdf output resolution is very low and unreadable compared to the source pdf file. The output pdf file size is also half the size of the original so it's doing something with the resolution. How do I keep the resolution/dpi the same as the original file?

**Comment by snibgo** [comment]
> How do I keep the resolution/dpi the same as the original file?

There is no "same resolution".

PDF files can contain vector and raster elements. Vector elements have no inherent resolution; they can be viewed at any scale with no pixelation.

IM is a raster image processor. When reading a PDF, it rasterizes each page, ie converts it to pixels. The resolution depends on the "-density" setting before reading the file. The default is "-density 72". Use 100, or 144, or 300, or whatever you want.

**Reply by ntec1** [reply]
Where do I go to change the density? Sorry I’m new to all this.

**Comment by snibgo** [comment]
For example:
```
magick -density 144 in.pdf out.png
```
What do you actually want to do? You seem to be reading PDF and writing PDF. Why? What processing are you doing?

Personally, I think PDF files are poor containers for raster images. They just add a layer of complexity for little or no benefit.

**Reply by ntec1** [reply]
It is installed for use on our word press web site. There is a section on our web site that allows a customer to upload a file which could be in a jpg, tif, png or pdf format to a server.

**Comment by mars0i** [comment]
Why? For example my library makes some books available as chapter PDFs, but I prefer to have a single PDF, and you can find advice on the web that recommends combining PDFs using `magick` or `convert`.  ImageMagick is an amazing tool, but as suggested above, combining PDFs is not really what it's for.  So for anyone wandering here looking to combine PDFs, I had better success calling ghostscript directly: `gs -dNOPAUSE -sDEVICE=pdfwrite -sOUTPUTFILE=combine.pdf -dBATCH 1.pdf 2.pdf`.
